### ISSUE TITLE
Test failure tracing/eventpipe/gcdump/gcdump/gcdump.sh

### ISSUE THREAD

**Issue by VincentBu** [dotnet/runtime#61824]
Run: [runtime-coreclr jitstress 20211117.1](https://dev.azure.com/dnceng/public/_build/results?buildId=1475284&view=ms.vss-test-web.build-test-results-tab&runId=42280206&resultId=110261&paneView=debug)

Failed test:
```
CoreCLR Linux arm64 Checked jitstress1_tiered @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-20210531091519-97d8652

- tracing/eventpipe/gcdump/gcdump/gcdump.sh
- tracing/eventpipe/eventsourceerror/eventsourceerror/eventsourceerror.sh
```
**Error message:**
```
Assert failure(PID 3009 [0x00000bc1], Thread: 3072 [0x0c00]): Assertion failed 'encoding_found' in 'Microsoft.Diagnostics.Tracing.EventPipeEventMetaDataHeader:ClearMemory(long,int)' during 'Emit code' (IL size 21)

File: /__w/1/s/src/coreclr/jit/emitarm64.cpp Line: 2203
Image: /root/helix/work/correlation/corerun
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9ad924a4)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9ad9d42c)
apply_reg_state: ip and cfa unchanged; stopping here (ip=0x7f9ad9d42c)
/root/helix/work/workitem/e/tracing/eventpipe/gcdump/gcdump/gcdump.sh: line 384:  3009 Aborted                 (core dumped) $LAUNCHER $ExePath "${CLRTestExecutionArguments[@]}"

Return code:      1
Raw output file:      /root/helix/work/workitem/uploads/Reports/tracing.eventpipe/gcdump/gcdump/gcdump.output.txt
Raw output:
BEGIN EXECUTION
/root/helix/work/correlation/corerun -p System.Reflection.Metadata.MetadataUpdater.IsSupported=false gcdump.dll ''
0.0s: ==TEST STARTING==
2.2s: Started sending sentinel events...
2.2s: Connecting to EventPipe...
2.8s: Connected to EventPipe with sessionID '0x7f1c0092b0'
2.8s: Creating EventPipeEventSource...
3.0s: EventPipeEventSource created
3.1s: Dynamic.All callback registered
3.1s: Running optional trace validator
3.5s: Finished running optional trace validator
3.5s: Starting stream processing...
Gathering state for process 3009 corerun
Crashing thread 00000c00 signal 00000006
Writing minidump with heap to file /home/helixbot/dotnetbuild/dumps/coredump.3009.dmp
Written 143093760 bytes (34935 pages) to core file
Dump successfully written
Expected: 100
Actual: 134
END EXECUTION - FAILED
Test Harness Exitcode is : 1
To run the test:

set CORE_ROOT=/root/helix/work/correlation
/root/helix/work/workitem/e/tracing/eventpipe/gcdump/gcdump/gcdump.sh
Expected: True
Actual:   False


Stack trace
   at tracing_eventpipe._gcdump_gcdump_gcdump_._gcdump_gcdump_gcdump_sh()
```

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib Looks like a regression (assertion) in arm64 encoding

**Comment by kunalspathak** [comment]
Fixed in #62106
